sql/expression/convert.go

```diff
@@ -210,12 +252,18 @@ func convertValue(val interface{}, castTo string, originType sql.Type) (interfac
 			}
 			b = encodedBytes
 		}
+		if typeLength > 0 {
```

Comment:
Should check that `typeLength <= len(b.([]byte))` to prevent OutOfBounds.

The query `SELECTCAST(123 AS BINARY(30));` panics.

Reply:
oooh... very good catch! I'll fix and add a test, too.